boutproject/BOUT-dev
# Fix some HDF5 related issues and add Mesh::getLocal{X,Y,Z}Index

Fixes #2003 

Also:

- Fix bug in `DataFormat::writeFieldAttributes` with `FieldPerp`: need to use the global index that doesn't include boundaries
- Fix `test-io_hdf5` needing global `mesh/dump`
- Build with HDF5 on Travis
- Require netCDF for some tests that use grid files
- Add Mesh::getLocal{X,Y,Z}Index{NoBoundaries} methods: these are inverses of the `getGlobal` versions, replacing `{X,Y}LOCAL`, and are necessary for consistent, unique global indices into the boundaries for double-null tokamak grids

There are some tests that can't use HDF5 because they use grid files, and I don't think we want to have duplicate binaries just in different formats. There are some other tests that could use HDF5 but currently can't because we'd need to be able to change the file extension in the `runtest` script. And some others which should be run with both HDF5 and netCDF, including some duplicated ones that could be merged, if we could query what formats are available in `runtest`. Not sure how much I actually want to do there, I'm not terribly certain we want to support HDF5 into the future.

Fedora test is failing because it needs `h5py`, but I didn't know where to specify that -- @dschwoerer does it need to be in your BOUT++ Fedora recipe? If so, is it possible to give me maintainer access to that?

Some of this should be backported.
